src/Shared/UrlDecoder/UrlDecoder.cs

```diff
@@ -307,11 +306,11 @@ private static int ReadHex(ref int scan, Span<byte> buffer)
             }
 
             var value = buffer[scan++];
-            var isHead = ((value >= '0') && (value <= '9')) ||
+            var isHex = ((value >= '0') && (value <= '9')) ||
```

Comment:
Lots of hex conversion utilities (including a very fast `IsHexChar` method) at https://github.com/dotnet/runtime/blob/main/src/libraries/Common/src/System/HexConverter.cs. Feel free to copy if you need them.

Reply:
Thanks, I will take a look and grab :)